rllib/evaluation/rollout_worker.py

```diff
@@ -533,7 +533,8 @@ def sample(self):
             batch = self.input_reader.next()
             steps_so_far += batch.count
             batches.append(batch)
-        batch = batches[0].concat_samples(batches)
+        batch = batches[0].concat_samples(batches).slice(start=0,
```

Comment:
This will discard data in other modes, so I don't think it can be added. What is the bug with TensorFlow, is there an issue in the tf repo for this?

Reply:
Hi, the length sampled data in offline training may vary and should be exactly equal to `rollout_fragment_length` when using `tf.function` (eager tracing) because the varying input shape (batch size) can incur increasing memory and lead to OOM.  Another workaround maybe adding callbacks

```
    def on_sample_end(self, worker: RolloutWorker, samples: SampleBatch,
                      **kwargs):
        for k in samples.data.keys():
            samples.data[k] = samples.data[k][0: worker.rollout_fragment_length]
```

https://github.com/tensorflow/tensorflow/issues/31842#issuecomment-523590836